src/libmongoc/tests/test-mongoc-sample-commands.c

```diff
@@ -2474,6 +2474,182 @@ test_example_56 (mongoc_database_t *db)
+   const bson_t *doc = NULL;
+   bson_error_t error;
+   const bson_value_t *value = NULL;
+   bson_iter_t iter;
+
+   pipeline = BCON_NEW ("pipeline",
+                        "[",
+                        "{",
+                        "$match",
+                        "{",
+                        BCON_UTF8 ("adoptable"),
+                        BCON_BOOL ("true"),
+                        "}",
+                        "}",
+                        "{",
+                        "$count",
+                        BCON_UTF8 ("adoptableCount"),
+                        "}",
+                        "]");
+
+   cursor = mongoc_collection_aggregate (
+      collection, MONGOC_QUERY_NONE, pipeline, NULL, NULL);
+
+   rc = mongoc_cursor_next (cursor, &doc);
+
+   if (mongoc_cursor_error (cursor, &error)) {
+      MONGOC_ERROR ("could not get adoptableCount: %s", error.message);
+      rc = false;
+      goto cleanup;
+   }
+
+   if (!rc) {
+      MONGOC_ERROR ("%s", "cursor has no results");
+      goto cleanup;
+   }
+
+   rc = bson_iter_init_find (&iter, doc, "adoptableCount");
+   if (rc) {
+      value = bson_iter_value (&iter);
+      if (value->value_type == BSON_TYPE_INT32) {
+         *accumulator += value->value.v_int32;
+      } else if (value->value_type == BSON_TYPE_INT32) {
+         *accumulator += value->value.v_int64;
+      } else {
+         MONGOC_ERROR ("%s", "'adoptableCount' must be an integer");
+         rc = false;
+         goto cleanup;
+      }
+   } else {
+      MONGOC_ERROR ("%s", "missing key: 'adoptableCount'");
+      goto cleanup;
+   }
+
+cleanup:
+   bson_destroy (pipeline);
+   mongoc_cursor_destroy (cursor);
+   return rc;
+}
+
+/*
+ * JIRA: https://jira.mongodb.org/browse/DRIVERS-2181
+ */
+static void
+test_example_59 (mongoc_database_t *db)
+{
+   /* Start Example 59 */
+   mongoc_client_t *client = NULL;
+   mongoc_client_session_t *cs = NULL;
+   mongoc_collection_t *cats_collection = NULL;
+   mongoc_collection_t *dogs_collection = NULL;
+   long long adoptable_pets_count = 0;
+   bson_error_t error;
+   mongoc_session_opt_t *session_opts;
+
+   client = test_framework_new_default_client ();
+
+   cats_collection = mongoc_client_get_collection (client, "pets", "cats");
+   dogs_collection = mongoc_client_get_collection (client, "pets", "dogs");
+
+   /* Seed 'pets.cats' and 'pets.dogs' with example data */
+   if (!pet_setup (cats_collection, dogs_collection)) {
+      goto cleanup;
+   }
+
+   /* start a snapshot session */
+   session_opts = mongoc_session_opts_new ();
+   mongoc_session_opts_set_snapshot (session_opts, true);
+   cs = mongoc_client_start_session (client, session_opts, &error);
+   mongoc_session_opts_destroy (session_opts);
+   if (!cs) {
+      MONGOC_ERROR ("Could not start session: %s", error.message);
+      goto cleanup;
+   }
+
+   accumulate_adoptable_count (cats_collection, &adoptable_pets_count);
+   accumulate_adoptable_count (dogs_collection, &adoptable_pets_count);
+
```

Comment:
It might be helpful to add a comment here describing that we are using an aggregation pipeline to return the count, since other drivers have been displaying the aggregation pipeline command used. Since this code example is already longer than other drivers, I think a comment is ok.

Reply:
I'm glad you brought this up, because that's something I'm curious about. It would be a fair amount of duplicate code and a bit lengthy to inline both calls to `accumulate_adoptable_count` in this function, but I'm wondering if I should do one of the following:

1. Include the entire function `accumulate_adoptable_count` in the example, i.e., place the start example comment just above the declaration of `accumulate_adoptable_count`, and end example comment stays where it is within `test_example_59`.
2. Place the text of `accumulate_adoptable_count` in a comment in `test_example_59`.
3. Explain in a comment what is happening in the function `accumulate_adoptable_count` purely in English, and omit code in the comment.
4. Inline all the code from `accumulate_adoptable_count` within `test_example_59`.

I'm leaning towards 1., but I'm not sure how that would look when rendered. I'd like to know what everyone thinks about this, since I'm not sure which way would be the cleanest for the purpose of producing an example.

Reply:
P.S. I'm planning on reaching out to Jeffrey Allen, the reporter for the original ticket, regarding this before merging.

Reply:
I got some feed back from Jeffery Allen. It sounds like he's in favor of option 3.

(quote from Jeff)
>Hey Kyle, thanks for reaching out. So my recollection with that ticket is that the main goal is to emphasize the snapshot functionality in the example. Therefore, I'm inclined to say that we can exclude the agg pipeline from the example, and leave a comment explaining the aggregation. That way, users aren't too overwhelmed by a large example and they'll still be able to see the snapshot behavior in action

Reply:
I agree option 3 would be the clearest, but I'm also not sure about how strict the formatting constraints are. Maybe the docs team could help make the choice.

Reply:
In [this page](https://www.mongodb.com/docs/v6.0/core/transactions/) the C example is much longer than others. So it could potentially be ok.

Reply:
I just saw the addition of the comment! Ignore my comment above.

Reply:
That's interesting. I do like when an example is explicit, since this example includes 2 functions, then I suppose no reason why I can't either. I'll make that change for now, and request another review from both you and Kevin to see what you think.

Reply:
Defer to Jeff's judgement since this example is requested by the documentation team.